### PR TITLE
Example for creating Workflow with nodepool Node Selectors and Tolerations

### PR DESCRIPTION
**Pull request by BnJam**
### What changes were proposed in this pull request?

I have included an extension on the DAG example that includes adding Node Selector and Tolerations when setting
up the Workflow to be submitted. I had to search through the codebase in order to build solution, so having an example
for others to reference in the future would be helpful. 

### Why are the changes needed?

The addition was needed to help end-users build effective Workflows in their environment if the submission 
nodepool is not the default. I had to search through the codebase in order to build the solution, so this example file
acts as a shortcut for future reference. 

### Does this PR introduce _any_ user-facing change?

This PR creates an additional example to add a Node Selector and Toleration to the Job Container. 

### How was this patch tested?

Couler was installed with the following command: 
`python3 -m pip install git+https://github.com/couler-proj/couler --ignore-installed` 

This example was tested on AKS via JupyterHub v1.2.0 to Argo Workflows v3.2.9 Helm deployment. As long as the tester/user is able to submit and process jobs, there is nothing else to set up for the example file since it is using the  unchanged core API.

### CODE REVIEW

File: examples/dag_node_assign.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        [
+            [lambda: job(name="A")],
+            [lambda: job(name="A"), lambda: job(name="B")],  # A -> B
+            [lambda: job(name="A"), lambda: job(name="C")],  # A -> C
+            [lambda: job(name="B"), lambda: job(name="D")],  # B -> D
+            [lambda: job(name="C"), lambda: job(name="D")],  # C -> D
+        ]
+    )
+
+
```
Comment (terrytangyuan) [inline examples/dag_node_assign.py:51]: These are not necessary to demonstrate node pool?

Comment (BnJam) [reply examples/dag_node_assign.py:51]: They are not, they are carried over from the DAG example as a self-contained example to submit

Comment (terrytangyuan) [reply examples/dag_node_assign.py:51]: Do you mind removing them and keep the example minimal?

**Review by terrytangyuan — APPROVED** [review]

**Comment by terrytangyuan** [comment]
Please fix the build. Thanks

**Comment by BnJam** [comment]
> Please fix the build. Thanks

Referenced your PR to fix build
https://github.com/argoproj/argo-workflows/pull/8254

**Comment by terrytangyuan** [comment]
Thanks please fix the CLA and semantic check as well.

**Pull request merged**